diff --git a/KindleReaderApiTestsCommon/build.gradle b/KindleReaderApiTestsCommon/build.gradle
index e7221f242a..35f978eac1 100644
--- a/KindleReaderApiTestsCommon/build.gradle
+++ b/KindleReaderApiTestsCommon/build.gradle
@@ -52,6 +52,9 @@ dependencies {
     implementation("com.amazon.kindle.instrumented.testngrunner:instrumentedtestngrunner:latest.integration")
     implementation project(path: ':LargeLibraryImplementation')
     implementation project(':TTSCore')
+    implementation libs.aae.espresso.sdk
+    implementation libs.androidx.ui.test.junit4.android
+    implementation libs.androidx.material3.android
     compileOnly("com.amazon.brazil.gradle.export:Phoenix-SDKModule-mapclientlib-62.0:latest.integration")
     implementation("com.amazon.kindle:AndroidRenderingModule:latest.integration")
     implementation("com.amazon.kindle:YJRenderingModule:latest.integration")
@@ -65,7 +68,8 @@ dependencies {
         exclude group: 'com.amazon.kindle', module: 'KindleReaderLibrary'
     }
     implementation(libs.universal.audio.player)
-
+    androidTestImplementation("androidx.compose.ui:ui-test:1.5.0")
+    debugImplementation("androidx.compose.ui:ui-test-manifest:1.5.0")
     testImplementation("org.mockito:mockito-core:1.9.0")
     testImplementation("com.amazon.brazil.gradle.export:PowerMockEasyMock-2.x:latest.integration")
     testImplementation(libs.bundles.powermock)
diff --git a/KindleReaderApiTestsCommon/configuration/yeTests.xml b/KindleReaderApiTestsCommon/configuration/yeTests.xml
new file mode 100644
index 0000000000..67a3afbc91
--- /dev/null
+++ b/KindleReaderApiTestsCommon/configuration/yeTests.xml
@@ -0,0 +1,24 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
+<suite name="sample" verbose="1" configfailurepolicy="continue">
+    <test name="Sample_tests">
+        <classes>
+            <class name="com.amazon.kindle.test.apie.SampleTests">
+                <methods>
+                    <include name="validateMultipleTabs" />
+                </methods>
+            </class>
+        </classes>
+    </test>
+<!--    <test name="Dictionary Tests - YJLR">-->
+<!--        &lt;!&ndash; This Year Will Be Different - YJLR &ndash;&gt;-->
+<!--        <parameter name="asin" value="B00TKWAV0S" />-->
+<!--        <classes>-->
+<!--            <class name="com.amazon.kindle.test.apie.SampleTests">-->
+<!--                <methods>-->
+<!--                    <include name="testDictionaryInfoCard" />-->
+<!--                </methods>-->
+<!--            </class>-->
+<!--        </classes>-->
+<!--    </test>-->
+</suite>
\ No newline at end of file
diff --git a/KindleReaderApiTestsCommon/src/com/amazon/kindle/test/apie/SampleTests.java b/KindleReaderApiTestsCommon/src/com/amazon/kindle/test/apie/SampleTests.java
new file mode 100644
index 0000000000..f506121a4c
--- /dev/null
+++ b/KindleReaderApiTestsCommon/src/com/amazon/kindle/test/apie/SampleTests.java
@@ -0,0 +1,115 @@
+package com.amazon.kindle.test.apie;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.amazon.a11y.aae.sdk.espresso.config.AaeMobileScanConfiguration;
+import com.amazon.a11y.aae.sdk.espresso.core.AaeMobileSdk;
+import com.amazon.a11y.aae.sdk.espresso.enums.ResourceType;
+import com.amazon.a11y.aae.sdk.espresso.enums.ScanSource;
+import com.amazon.kindle.test.KcpTestRunner.framework.logging.KcpTestLogger;
+import com.amazon.kindle.test.KrxTestSDKProvider;
+import com.amazon.kindle.test.api.robots.outofbook.RubyLibraryRobot;
+import com.amazon.kindle.test.api.util.ContentUtils;
+import com.amazon.kindle.test.api.util.InBookActions;
+import com.amazon.kindle.test.api.util.SelectionMenuUtils;
+import com.amazon.kindle.test.framework.logging.TestNGLogger;
+
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+
+
+public class SampleTests extends InBookBase {
+    public String contentFormat;
+    private RubyLibraryRobot libraryRobot;
+    private AaeMobileSdk sdk;
+    private String asin;
+
+//    @Parameters({"asin"})
+//    public SampleTests(String asin) {
+//        this.asin = asin;
+//    }
+
+    @BeforeMethod(alwaysRun = true)
+    public void methodSetup() {
+        this.libraryRobot = RubyLibraryRobot.getInstance();
+        krxTestSDK.getHomeRobot().goToHome();
+        TestNGLogger.log("Create AAE config and sdk");
+        var instrumentation = InstrumentationRegistry.getInstrumentation();
+        // Base configuration
+        var config = AaeMobileScanConfiguration.builder()
+                .resourceId("com.amazon.kindle")
+                .resourceType(ResourceType.Package)
+                .scanSource(ScanSource.Dev)
+                // Path is relative to the app's storage directory
+                .artifactsLocalStoragePath("/files/kindle/a11yReport")
+                .instrumentation(instrumentation)
+                .build();
+        // Create SDK with base config
+        sdk = new AaeMobileSdk(config);
+        //sdk.startContinuousScan();
+//        super.setupClass(asin);
+//        this.contentMetadata = ContentUtils.getContentMetadata(testContent);
+
+    }
+
+//    @Test
+//    public void validateLibraryTab() throws InterruptedException {
+//        TestNGLogger.log("Navigating to Library");
+//        libraryRobot.clickOnLibraryTab();
+//        Thread.sleep(1500);
+//
+//
+//        TestNGLogger.log("Use sdk to perform Single Screen Scan.");
+//        try {
+//            sdk.performSingleScreenScanAndReport();
+//        } catch (IllegalStateException e) {
+//            // Handle configuration or setup errors
+//            TestNGLogger.log("Failed to initialize scan:" + Arrays.toString(e.getStackTrace()));
+//        } catch (RuntimeException e) {
+//            // Handle scan execution or report generation errors
+//            TestNGLogger.log("Scan failed", e);
+//        }
+//    }
+
+    @Test
+    public void validateMultipleTabs() throws InterruptedException {
+//        TestNGLogger.log("Navigating to Home");
+//        libraryRobot.clickOnHomeTab();
+        TestNGLogger.log("Navigating to Library");
+        libraryRobot.clickOnLibraryTab();
+        Thread.sleep(20000);
+        sdk.performSingleScreenScanAndReport();
+    }
+
+
+
+    @Test
+    public void testDictionaryInfoCard() throws Throwable {
+        TestNGLogger.log("Opening and downloading the dictionary for first time");
+        ContentUtils.ensureBookOpenAndChromeClosed(testContent, contentMetadata);
+        InstrumentationRegistry.getInstrumentation().waitForIdleSync();
+        KcpTestLogger.LogDebugMessage("Opening chrome bar in " + this.testContent.getAsin().toString());
+        InBookActions.getInstance().openChrome();
+        Thread.sleep(10000);
+        sdk.performSingleScreenScanAndReport();
+//        KrxTestSDKProvider.get().getNavigationRobot().navigateToLocation(location);
+//        SelectionMenuUtils.select(word, 1);
+    }
+
+//    @AfterMethod(alwaysRun = true)
+//    public void methodCleanUp() {
+//        TestNGLogger.log("Stop Scan");
+//        if (sdk != null) {
+//            try {
+//                // Ensure continuous scans are stopped
+//                sdk.stopContinuousScan();
+//            } catch (IllegalStateException ignored) {
+//                // Ignore if no scan was running
+//            }
+//        }
+//    }
+}
\ No newline at end of file
diff --git a/StandAlone/build.gradle.kts b/StandAlone/build.gradle.kts
index f8633302d1..6270faf1d3 100644
--- a/StandAlone/build.gradle.kts
+++ b/StandAlone/build.gradle.kts
@@ -663,6 +663,13 @@ dependencies {
 
     implementation("com.amazon.falkor:FalkorAndroidPlugin:latest.integration")
 
+    api("com.amazon.a11y.aae.sdk.espresso:aae-espresso-sdk:1.0.14")
+    androidTestImplementation("androidx.test:core:1.5.0")
+    androidTestImplementation("androidx.test.ext:junit:1.2.1")
+    androidTestImplementation("androidx.compose.ui:ui-test:1.5.0")
+    debugImplementation("androidx.compose.ui:ui-test-manifest:1.5.0")
+
+
     implementation(libs.socialsharing.thirdparty)
     implementation("com.facebook:facebook-android-api:${libs.versions.facebookApi.get()}")
 
diff --git a/brazil.ion b/brazil.ion
index f984a571a1..be5903a197 100644
--- a/brazil.ion
+++ b/brazil.ion
@@ -79,6 +79,7 @@ common::{
             "KindleLearningObjectModule",
             "KindleAndroidKeduFTUE",
             "FreeTimeBookPlugin",
+            "AaeEspressoSdk",
         ]
     },
 
diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index 91cff124ad..7264f59855 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -176,6 +176,9 @@ androidx-material3-window-size = { group = "androidx.compose.material3", name =
 
 whispersync-clientlib = { group = "com.amazon.device.sync", name = "WhispersyncClientLib", version.ref = "whispersyncClientLib" }
 whispersync-phoenix = { group = "com.amazon.fireos.phoenixsdk", name = "whispersync", version.ref = "whispersync" }
+aae-espresso-sdk = { group = "com.amazon.a11y.aae.sdk.espresso", name = "aae-espresso-sdk", version.ref = "aaeEspressoSdk" }
+androidx-ui-test-junit4-android = { group = "androidx.compose.ui", name = "ui-test-junit4-android", version.ref = "uiTestJunit4Android" }
+androidx-material3-android = { group = "androidx.compose.material3", name = "material3-android", version.ref = "material3Android" }
 
 kids-plugin = { group = "com.amazon.tahoe.book", name ="FreeTimeBookPlugin", version.ref = "peru" }
 
@@ -326,3 +329,10 @@ junit4 = "4.13.2"
 powermock = "2.0.0"
 robolectric = "4.13"
 sqliteJdbc = "3.7.2"
+
+uiAndroid = "1.6.1"
+uiToolingPreviewAndroid = "1.7.7"
+
+aaeEspressoSdk = "1.0.14"
+uiTestJunit4Android = "1.7.8"
+material3Android = "1.3.1"
