diff --git a/src/elk_test_runner/testng.py b/src/elk_test_runner/testng.py
index be88ea8..3018202 100755
--- a/src/elk_test_runner/testng.py
+++ b/src/elk_test_runner/testng.py
@@ -729,7 +729,7 @@ class ApiTestHarness:
             output = self.guarded_adb_shell(f"ls {self.output_files.testng_path}")
             if output and output[0] != "":
                 self.logger.info("Found previous runs on the sdcard, removing them")
-                delete_path_from_device(self.dsn, self.output_files.testng_path)
+                # delete_path_from_device(self.dsn, self.output_files.testng_path)
         except subprocess.CalledProcessError as e:
             warn("CalledProcessError: {}", e)
         except Exception:
@@ -772,10 +772,11 @@ class ApiTestHarness:
         """
         Load testNG results file
         """
-        self.logger.info(f"Pushing {self.test_xml} to {self.output_files.testng_path}")
-        self.guarded_adb_shell(f"chmod 777 {self.output_files.testng_path}")
-        self.guarded_adb(["push", self.test_xml, self.output_files.testng_path / self.test_xml.name])
-        self.guarded_adb_shell(f"chmod 777 {self.output_files.testng_path}/{self.test_xml.name}")
+        # self.logger.info(f"Pushing {self.test_xml} to {self.output_files.testng_path}")
+        # self.guarded_adb_shell(f"chmod 777 {self.output_files.testng_path}")
+        # self.guarded_adb(["push", self.test_xml, self.output_files.testng_path / self.test_xml.name])
+        # self.guarded_adb_shell(f"chmod 777 {self.output_files.testng_path}/{self.test_xml.name}")
+        pass
 
     def clear_app_data(self):
         """
